python/cudf/dataframe/dataframe.py

```diff
@@ -1889,6 +1889,9 @@ def query(self, expr, local_dict={}):
                         datetimes
         1 2018-10-08T00:00:00.000
         """
+        if self.empty:
+            return self.copy()
```

Comment:
Not sure, if we prefer a deep copy here. Should not really matter as empty data frame.

Reply:
self.copy() defaults to a deep copy.

Reply:
Aah, got it. I was mistaken. Thanks for clarifying. 
I think everything else should be fine to merge.